resources/knative-eventing/charts/knative-eventing/templates/uninstall-eventing.yaml

```diff
@@ -0,0 +1,144 @@
+    resources: ["*"]
+    verbs: ["list", "delete"]
+  - apiGroups: ["apiextensions.k8s.io"]
+    resources: ["customresourcedefinitions"]
+    verbs: ["list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ .Release.Name }}-cr-delete
+  annotations:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    helm.sh/hook-weight: "0"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: {{ .Release.Name }}-cr-delete
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Release.Name }}-cr-delete
+    namespace: {{ .Release.Namespace }}
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ .Release.Name }}-cr-delete
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    helm.sh/hook: pre-delete
+    helm.sh/hook-delete-policy: "hook-succeeded,before-hook-creation"
+    sidecar.istio.io/inject: "false"
+    # dependent on clusterrole, clusterrolebinding, serviceaccount
+    # => hook-weight=min(hook-weight)+1
+    helm.sh/hook-weight: "1"
+spec:
+  backoffLimit: 1
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+      name: {{ .Release.Name }}-cr-delete
+    spec:
+      serviceAccountName: {{ .Release.Name }}-cr-delete
+      restartPolicy: Never
+      containers:
+        - name: {{ .Release.Name }}-cr-delete
+          image: eu.gcr.io/kyma-project/test-infra/alpine-kubectl:v20200507-070ff576
+          command:
+            - "/bin/bash"
+          args:
+            - "-c"
+            - |
+              set -eux -o pipefail
+              echo "starting deletion of knative-eventing core custom resources"
+              resources=(
+                "apiserversources.sources.eventing.knative.dev"
+                "containersources.sources.eventing.knative.dev"
+                "cronjobsources.sources.eventing.knative.dev"
+                "sinkbindings.sources.eventing.knative.dev"
+
+                "eventtypes.eventing.knative.dev"
+                "brokers.eventing.knative.dev"
+                "triggers.eventing.knative.dev"
+
+                "channels.messaging.knative.dev"
+                "parallels.messaging.knative.dev"
+                "sequences.messaging.knative.dev"
+                "subscriptions.messaging.knative.dev"
+              )
+
+              for resource in "${resources[@]}"; do
+                echo "starting deletion of ${resource} custom resources"
+                if kubectl get crd "${resource}"; then
+                  kubectl \
+                    delete \
+                    --ignore-not-found \
+                    "${resource}" \
+                    --all-namespaces \
+                    --all
+                else
+                  echo "CRD ${resource} unknown to apiserver"
+                fi
+                echo "finished deletion of ${resource} custom resources"
+              done
+
+              echo "done"
```

Comment:
i think it is possible to read template content from files... that way it would be quite easy to test such scripts locally

Reply:
https://helm.sh/docs/chart_template_guide/accessing_files/

Reply:
Jup, we can do so.